pkg/cmd/admin/router/router.go

```diff
@@ -661,6 +650,17 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 		if errs := bulk.Create(list, namespace); len(errs) != 0 {
 			return errExit
 		}
+
+		if needsSCC {
+			fmt.Fprintf(os.Stderr, "WARNING: Router service account %q is not privileged to open ports on the host IP\n", cfg.ServiceAccount)
+		}
+
+		if sobjects, err := associateSecrets(kClient, secrets, namespace, cfg.ServiceAccount, objects); err != nil {
+			return err
+		} else {
+			objects = sobjects
```

Comment:
Not for now (it can be addressed later) - but imho, it feels cleaner to not overwrite `objects` but rather have `associateSecrets` (or something more appropriately named) return a list of objects and iterate over those returned entities (objects seems to be an overloaded/overused term here!!) and add 'em to `objects`. Potential to clobber stuff exists if the function does something wonky with parameter `objects`.